From 35c212ea82a37cfb35ea81c9b66d4c08090622c2 Mon Sep 17 00:00:00 2001
From: Sean Cross <xobs@kosagi.com>
Date: Mon, 16 Nov 2015 14:12:26 +0800
Subject: [PATCH 20/65] package-kernel: Add our own package script

This script is used by Kosagi to package kernels for Novena.

Signed-off-by: Sean Cross <xobs@kosagi.com>

(sakaki: patchlevel 1 -> 0, as required by Gentoo's unipatch)
---
 package-kernel.sh | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)
 create mode 100755 package-kernel.sh

diff --git a/package-kernel.sh b/package-kernel.sh
new file mode 100755
index 0000000..5d1c2ab
--- /dev/null
+++ package-kernel.sh
@@ -0,0 +1,34 @@
+#!/bin/sh
+# A script to build the kernel .deb files, as used by Kosagi.
+threads=$(($(grep '^processor' /proc/cpuinfo  | cut -d: -f2 | tail -n1)+1))
+version=4.4-novena-r1
+
+echo "Building v${version} with ${threads} threads"
+git tag -d "v${version}" 2> /dev/null
+git tag "v${version}"
+make -j${threads} \
+	KBUILD_DEBARCH=armhf \
+	KBUILD_IMAGE=zImage \
+	KBUILD_DTB=imx6q-novena.dtb \
+	KBUILD_DESTDIR=usr/share/linux-novena \
+	KDEB_PKGVERSION=${version} \
+	KDEB_PKGNAME="novena" \
+	KDEB_SOURCENAME="linux-image-novena" \
+	EMAIL="xobs@kosagi.com" \
+	NAME="Sean Cross" \
+	dtbs || exit 1
+
+# Delete the debian "files" listing, as it tends to be out of date
+rm -f debian/files
+
+make -j${threads} \
+	KBUILD_DEBARCH=armhf \
+	KBUILD_IMAGE=zImage \
+	KBUILD_DTB=imx6q-novena.dtb \
+	KBUILD_DESTDIR=usr/share/linux-novena \
+	KDEB_PKGVERSION=${version} \
+	KDEB_PKGNAME="novena" \
+	KDEB_SOURCENAME="linux-image-novena" \
+	EMAIL="xobs@kosagi.com" \
+	NAME="Sean Cross" \
+	deb-pkg
-- 
2.7.3

